### DISCUSSION TITLE
Class is not using lazy loading, eventhough created instance of an image

### DISCUSSION THREAD

**Discussion by CaroDur** [TorchIO-project/torchio#589] [Q&A] [answered]
My list includes around 400 images with a memory usage on the disc of approx. 1 GB. When i run the following code and check RAM usage, it uses 55GB RAM. Which is obviously not lazy loading. 

```python
subjects = []
for (image_name, label_name) in zip(image_names, label_names):
    subject = tio.Subject(
        image=tio.ScalarImage(os.path.join(images_dir, image_name)),
        label=tio.LabelMap(os.path.join(labels_dir, label_name) ),
        vendor='contrast',
    )
    subjects.append(subject)
```

Can someone help me here or sees the problem? Thank you very much for the help and the great library :)
Greetings

**Comment by fepegar** [comment, marked as answer]
Hi, @CaroDur. Thanks for reporting. That shouldn't be happening. Is that all the code?

**Reply by CaroDur** [reply]
Well no, not entirely. I need to split my label in four channels, to have one channel per label. So my code to generate the list of subjects looks like that: (num_classes=4)
***************************************************************
subjects = []
for (image_name, label_name) in zip(image_names, label_names):
    subject = tio.Subject(
        image=tio.ScalarImage(os.path.join(images_dir, image_name)),
        label=tio.LabelMap(os.path.join(labels_dir, label_name) ),
        vendor='contrast',
    )

    labelTrain = torch.zeros((num_classes, subject['label']['data'].shape[1], subject['label']['data'].shape[2], subject['label']['data'].shape[3]))
    for i in range(num_classes):
        currentLabel = subject['label']['data']
        labelToAdd = torch.where(currentLabel == i, 1, 0)
        labelTrain[i,...] = labelToAdd
    
    subject.add_image(tio.LabelMap(tensor=labelTrain,  affine=subject.image.affine), 'labelTrain')
    subjects.append(subject)
    
***************************************************************************************

Thank you very much for your very fast reply! :)
Have a nice evening,
Carolina

**Reply by fepegar** [reply]
When you run `subject['label']['data']`, you are loading the data from the `label` image onto the RAM. That's why you're seeing that memory usage. If you don't need `label` for training, maybe you can use `subject.remove_image` to clear some memory. But `labelTrain` will still take some, as you are passing the tensor directly instead of a file path. Does that make sense?

PS: Please use Python [syntax highlighting](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting) for your code snippets.

**Comment by CaroDur** [comment]
Yes, that makes sense! Thank you very much. Didn't think of that.

Wish you a nice day!
